### ISSUE TITLE
GPU driver test fail in asus zenfone 2 (Adreno 506)

### ISSUE THREAD

**Issue by sum2012** [hrydgard/ppsspp#14109]
OpenGL
![Screenshot_20210211-085401](https://user-images.githubusercontent.com/2177532/107593382-36574200-6c4a-11eb-813a-cdc8b6149512.jpg)
![Screenshot_20210211-085408](https://user-images.githubusercontent.com/2177532/107593387-38b99c00-6c4a-11eb-8776-493807b4f5b4.jpg)
![Screenshot_20210211-085423](https://user-images.githubusercontent.com/2177532/107593389-39523280-6c4a-11eb-909a-b669a8baa468.jpg)
![Screenshot_20210211-085433](https://user-images.githubusercontent.com/2177532/107593390-39eac900-6c4a-11eb-8e05-488470e59ec4.jpg)
![Screenshot_20210211-085512](https://user-images.githubusercontent.com/2177532/107593391-3a835f80-6c4a-11eb-96dc-198c9e134dfd.jpg)

Vulkan
![Screenshot_20210211-090358](https://user-images.githubusercontent.com/2177532/107593464-56870100-6c4a-11eb-8f12-910404e873f1.jpg)
![Screenshot_20210211-090414](https://user-images.githubusercontent.com/2177532/107593468-5981f180-6c4a-11eb-8934-b505ae36e84b.jpg)
![Screenshot_20210211-090424](https://user-images.githubusercontent.com/2177532/107593470-5a1a8800-6c4a-11eb-963b-caea9b320985.jpg)
![Screenshot_20210211-090927](https://user-images.githubusercontent.com/2177532/107593473-5c7ce200-6c4a-11eb-9fbc-1f8b0de087dd.jpg)

**Comment by Panderner** [comment]
That phone model is Asus Zenfone 3 not Zenfone 2.

**Comment by sum2012** [comment]
Sorry,That phone is too old so that I remember wrong

**Comment by unknownbrackets** [comment]
Isn't this expected?  It's not really a bug in PPSSPP if it's correctly failing on devices/drivers that have the glitch.

-[Unknown]

**Comment by hrydgard** [comment]
The test is there to show that the GPU driver of the device has a bug. It's not a bug in PPSSPP. It can be useful though maybe to collect a list of devices that are affected, so I'll leave the thread open for that issue.

**Comment by sum2012** [comment]
Keep record Black shark 3 (Adreno 650) don't have same result on
https://user-images.githubusercontent.com/2177532/107593470-5a1a8800-6c4a-11eb-963b-caea9b320985.jpg

**Comment by sum2012** [comment]
@unknownbrackets
Do we have driver bug workaround on
 https://user-images.githubusercontent.com/2177532/107593391-3a835f80-6c4a-11eb-96dc-198c9e134dfd.jpg ?

**Comment by hrydgard** [comment]
We have some partial workarounds, but they're intentionally not used in the test, so that the test keeps showing that the GPU driver has a bug.

**Comment by kage2051** [comment]
Unfortunately all OGL drivers for Adreno expose that failed test. Which means OGL on mobile is pretty limited compared to PC. Below is a proof of that, on a Redmi Note 6 Pro after the update to MIUI 12.

![Screenshot_2021-02-11-10-24-09-277_org ppsspp ppsspp](https://user-images.githubusercontent.com/25725203/107668526-64428380-6c56-11eb-883c-485591032646.jpg)

**Comment by unknownbrackets** [comment]
Actually, I think we don't use any workaround on GLES, only on Vulkan.  It's possible there's a bug in the test on GLES, though... but it's not crucial as it was mainly aimed to give driver developers a way to fix the Vulkan driver (which they did.)

-[Unknown]

**Comment by Gamemulatorer** [comment]
OpenGL
![Screenshot_2022-09-14-05-24-19-37](https://user-images.githubusercontent.com/37603562/190012114-af207b44-1fa7-424a-a127-be8f513549f3.png)
![Screenshot_2022-09-14-05-24-23-42](https://user-images.githubusercontent.com/37603562/190012132-2cc7b2f9-90d6-42b9-aad9-108d7de8727a.png)

Vulkan
![Screenshot_2022-09-14-05-23-59-33](https://user-images.githubusercontent.com/37603562/190012190-455d8812-7bff-4a73-8a5c-7b4677935b16.png)
![Screenshot_2022-09-14-05-24-02-72](https://user-images.githubusercontent.com/37603562/190012210-818b074b-315f-43d4-baa4-af8394f0dd64.png)

**Comment by unknownbrackets** [comment]
I suspect we should just close this issue.  We do have a workaround for this driver bug.

It would be a bug in PPSSPP if your driver had the bug, but this test said "TEST OK."  So it's doing exactly it's job right now, and showing the results of your buggy driver.  That it also sees problems on OpenGL might be a bug in the test, I wonder if it's ending up with a backbuffer that lacks stencil or something.

-[Unknown]

**Comment by Gamemulatorer** [comment]
Vivo Y11 Snapdragon 439 Adreno 505
![Screenshot_20221026_012944_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37603562/197842417-7ba533ae-e890-411a-bb55-a2df63e17a61.jpg)
![Screenshot_20221026_012949_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37603562/197842422-f1c4578f-f7ed-4343-96ec-a954c4d8e264.jpg)

**Comment by hrydgard** [comment]
Good to see that the stencil test is fixed on that driver. Might be time to figure out at what Adreno driver version to cut off the workaround, though it's not very high impact so not that important.

Disappointing that Adreno still hasn't fixed that other simple discard bug though that I call "Adreno logic", it's quite ridiculous. I have reported it a long time ago... We are not affected fortunately at least after a cheap workaround I did.

**Comment by unknownbrackets** [comment]
Right, that bug was fixed a while back in the driver, see 8d900bb432db648d1378a37b2d340139536fd2e3.

That other Adreno logic bug always makes me worry about other conditionals.  Hopefully it's only when it's around a discard...

-[Unknown]

**Comment by Crisanto47** [comment]
Might be good to remove this also there's nothing useful about this feature.
